WordPressAuthenticator/Unified Auth/View Related/Site Address/SiteAddressViewController.swift

```diff
@@ -493,9 +493,9 @@ private extension SiteAddressViewController {
                 // It's not guaranteed to be included in the error object depending on the error.
                 DDLogInfo("Error attempting to connect to site address: \(self.loginFields.siteAddress)")
                 DDLogError(error.localizedDescription)
-                // TODO: - Tracks.
-                // WordPressAuthenticator.track(.loginFailedToGuessXMLRPC, error: error)
-                // WordPressAuthenticator.track(.loginFailed, error: error)
+
+                self.tracker.track(failure: .loginFailedToGuessXMLRPC)
```

Comment:
As I look at [the few lines below](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/c169358f383aa0bcbfb383dfa03ae983d543b266/WordPressAuthenticator/Unified%20Auth/View%20Related/Site%20Address/SiteAddressViewController.swift#L518-L531), I see that the error can be something other than XML-RPC issue. Should we reuse the `message` from those lines and log them when `authenticationDelegate.shouldHandleError` returns `true`?

Reply:
Sorry, I am trying to understand this. 

I believe the purpose of this event is to track the number of failures happen at this "guessing XMLRPC" step of the login process. 

> I see that the error can be something other than XML-RPC issue

If we eventually remove XMLRPC, those other issues wouldn't really matter to the WooCommerce app as long as we can establish communication with the REST API. (Jetpack tunnel) Please correct me if I am wrong. (i.e. WordPress mobile app can partially work with just XMLRPC so these errors would matter to them.)

> Should we reuse the message from those lines and log them when authenticationDelegate.shouldHandleError returns true?

Do you mean to keep this new tracking event and also add another tracking event for this case?

Reply:
Thanks for the detailed explanation and code. 😄 

Done in f2ad89e43f769ac144dcb19df89bb5c78efa7c2a